### PR TITLE
Run SAML Multi Auth integration tests in Cypress

### PR DESCRIPTION
**Pull request by cwperks**
### Description

Refactors the existing SAML Multi-auth test from Selenium to Cypress. This abstracts the 2 SAML login methods to `commands.js` and takes advantage of code re-use to run the same SAML test suite with different methods of logging in. 

To run the multiauth test suite for SAML its necessary to add `--env loginMethod=saml_multiauth` to the `cypress run` command.

```
yarn cypress:run --browser chrome --headless --env loginMethod=saml_multiauth --spec 'test/cypress/e2e/saml/*.js'
```

### Category

Refactor

### Issues Resolved

- https://github.com/opensearch-project/security-dashboards-plugin/issues/1667

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cwperks** [comment]
The failing checks will fail until https://github.com/opensearch-project/security-dashboards-plugin/pull/1728 is merged.

**Review by DarshitChanpura — DISMISSED** [review]

File: .github/workflows/cypress-test-multiauth-e2e.yml [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+              echo "Timeout for 600 seconds reached. OpenSearch Dashboards did not finish compiling."
+              exit 1
+            fi
+            sleep 10
+          done
+
+      - name: Run Cypress
+        run : |
+          yarn add cypress --save-dev
+          yarn cypress:run --browser chrome --headless --env loginMethod=saml_multiauth --spec 'test/cypress/e2e/saml/*.js'
```
Comment (peternied) [inline .github/workflows/cypress-test-multiauth-e2e.yml:210]: This file is 99% the same as `cypress-test-saml-e2e.yml`, lets refactor all these cypress workflows together.  If there are only 2 lines different, lets figure out how to bake these together.

I suspect that by writing a way to append to the opensearch_dashboards.yml then its just a matter of defining a matrix of environments.

**Review by peternied — CHANGES_REQUESTED** [review]

**Comment by cwperks** [comment]
@peternied This is now refactored to better take advantage of code re-use.

File: .github/workflows/cypress-test-multiauth-e2e.yml (hunk 1 above)
Comment (cwperks) [reply .github/workflows/cypress-test-multiauth-e2e.yml]: This is refactored into a reusable action now.

**Review by peternied — APPROVED** [review]
Nicely done!

**Review by DarshitChanpura — APPROVED** [review]
Nice work here @cwperks 👏 !

**Pull request merged**